subcommand/mesh-init/command_test.go

```diff
@@ -151,6 +191,51 @@ func TestRun(t *testing.T) {
 			envoyBootstrapContents, err := ioutil.ReadFile(envoyBootstrapFile.Name())
 			require.NoError(t, err)
 			require.NotEmpty(t, envoyBootstrapContents)
+
+			if c.checks != nil {
+				actualChecks, err := consulClient.Agent().Checks()
+				require.NoError(t, err)
+				for _, expCheck := range c.checks {
+					expectedAgentCheck := toAgentCheck(expCheck)
+					// Check for "critical" status. There is no listening application here, so checks will not pass.
+					expectedAgentCheck.Status = api.HealthCritical
+					// Pull the check type from the CheckID: "api-<type>" -> "<type>"
+					expectedAgentCheck.Type = strings.ReplaceAll(expCheck.CheckID, "api-", "")
```

Comment:
Why do we need to do that?

Reply:
Consul has a `type` field in the response, but not the request ([link to request examples](https://www.consul.io/docs/discovery/checks#check-definition)). 

So I encoded type into the check id. 

I guess I could make the check id be the exact check type to simplify. Or, I could include the expected check type as a separate field in the test case data.

Reply:
I'm going to merge as-is - but happy to follow up and change/simplify this if you think we should! 😀